mbed_targets/_internal/target_database.py

```diff
@@ -2,41 +2,65 @@
 
 import logging
 from json.decoder import JSONDecodeError
-from typing import List, Optional
+from typing import List
 
 import requests
 
 logger = logging.getLogger(__name__)
 
 _AUTH_TOKEN_ENV_VAR = "MBED_API_AUTH_TOKEN"
-
 _TARGET_API = "https://os.mbed.com/api/v4/targets"
 
 
-def get_target_data() -> Optional[List[dict]]:
+class TargetAPIError(Exception):
+    """API request failed."""
+
+
+class ResponseJSONError(Exception):
+    """HTTP reponse JSON parsing failed."""
```

Comment:
This raises an interesting point, we should have a well defined exception hierarchy which is shared by all modules with a base class of a tools error so that it can be sensible caught by a calling script.

Reply:
Good point. I have created an exception hierarchy in the latest commit. Let me know what you think.